azure-pipelines.yaml

```diff
@@ -0,0 +1,166 @@
+  # publish test results
+- task: PublishTestResults@2
+  displayName: Publish test results
+  inputs:
+    testResultsFormat: JUnit
+    testResultsFiles: test-output/junit.xml
+    failTaskOnFailedTests: true
+    testRunTitle: Unit Test
+  # deploy via helm
+- task: HelmDeploy@0
+  inputs:
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureResourceGroup: $(azureResourceGroup)
+    kubernetesCluster: $(kubernetesCluster)
+    namespace: $(imageName)-$(safeContainerTag)
+    command: upgrade
+    chartType: filepath
+    chartPath: helm
+    releaseName: $(imageName)-$(containerTag)
+    arguments: --set image=$(azureContainerRegistryFull)/$(imageName):$(containerTag) --set container.redeployOnChange=$(Build.BuildId) --set container.imagePullPolicy=Always --atomic
+    install: true
+    waitForExecution: true
+  displayName: Helm deploy
+  # delete helm chart of closed PR
+- task: HelmDeploy@0
+  condition: and(succeeded(), ne(variables['mergedPrNo'], ''))
+  inputs:
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureResourceGroup: $(azureResourceGroup)
+    kubernetesCluster: $(kubernetesCluster)
+    command: delete
+    arguments: --purge $(imageName)-pr$(mergedPrNo)
+  displayName: delete Helm chart of closed PR
+  # delete PR image from registry if merge to master
+- task: AzureCLI@1
+  condition: and(succeeded(), ne(variables['mergedPrNo'], ''))
+  inputs:
+    azureSubscription: $(azureSubscriptionEndpoint)
+    scriptLocation: inlineScript
+    inlineScript: |
+      az acr repository delete --name $(azureContainerRegistry) --image $(imageName):pr$(mergedPrNo) -y
+    failOnStandardError: false
+  displayName: remove merged PR image
+  # add Helm chart to registry for a release
+- task: AzureCLI@1
+  condition: and(succeeded(), contains(variables['Build.SourceBranch'], 'refs/tags'))
+  inputs:
+    azureSubscription: $(azureSubscriptionEndpoint)
+    scriptLocation: inlineScript
+    inlineScript: |
+      helm init -c
+      az configure --defaults acr=$(azureContainerRegistry)
+      az acr helm repo add
+      helm package ./helm --save
+      packageFile=`ls elm-rbac-* |  xargs echo`
+      az acr helm push --force $packageFile
+      az acr helm list
+    failOnStandardError: false
+  displayName: add Helm chart to registry
+# delete merged PR namespace from cluster
+- task: Kubernetes@1
+  condition: and(succeeded(), ne(variables['mergedPrNo'], ''))
+  inputs:
+    connectionType: Azure Resource Manager
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureResourceGroup: $(azureResourceGroup)
+    kubernetesCluster: $(kubernetesCluster)
+    command: delete
+    arguments: namespaces $(imageName)-pr$(mergedPrNo)
+  displayName: delete merged PR namespace
+  # display namepace of deployed build
+- script: |
+    echo "Build available for review in namespace $(imageName)-$(containerTag)"
+  displayName: display namespace of deployed build
```

Comment:
There needs to be some final tasks to install helm on the build agent, package the helm chart, extract the production values yaml and package all for the release pipeline.  eg.

```
  # install helm on build agent
- task: HelmInstaller@0
  displayName: 'Install Helm 2.9.1'
  inputs:
    helmVersion: 2.9.1
  # package helm chart for release
- task: HelmDeploy@0
  displayName: 'helm package'
  inputs:
    command: package
    chartPath: ./helm
    save: false
  # copy yaml values for release  
- task: CopyFiles@2
  displayName: 'Copy values files'
  inputs:
    SourceFolder: helm
    Contents: '*-values.yaml'
    TargetFolder: '$(Build.ArtifactStagingDirectory)'
  # publish artifact for release
- task: PublishBuildArtifacts@1
  displayName: 'Publish Artifacts: drop'
```

I think some of the earlier created pipelines are also missing this.

Reply:
Added in those tasks